model/lifecycle.go

```diff
@@ -1771,6 +1771,34 @@ func addNewTasks(ctx context.Context, creationInfo TaskCreationInfo, existingBui
 	return activatedTaskIds, nil
 }
 
+// activateExistingInactiveTasks will find existing inactive tasks in the patch that need to be activated as
+// part of the patch re-configuration.
+func activateExistingInactiveTasks(creationInfo TaskCreationInfo, existingBuilds []build.Build) error {
+	for _, b := range existingBuilds {
+		tasksInBuild, err := task.FindAll(db.Query(task.ByBuildId(b.Id)))
+		if err != nil {
+			return err
+		}
+		existingTasksIndex := map[string]task.Task{}
+		for i := range tasksInBuild {
+			existingTasksIndex[tasksInBuild[i].DisplayName] = tasksInBuild[i]
+		}
+		existingTasksToActivate := []task.Task{}
+		execAndDisplayTasks := append(creationInfo.Pairs.ExecTasks.TaskNames(b.BuildVariant), creationInfo.Pairs.DisplayTasks.TaskNames(b.BuildVariant)...)
+		for _, taskName := range execAndDisplayTasks {
+			if t, ok := existingTasksIndex[taskName]; ok && !t.Activated {
+				existingTasksToActivate = append(existingTasksToActivate, t)
+			}
+		}
+		if len(existingTasksToActivate) > 0 {
```

Comment:
Can we call this just once outside the for loop since it doesn't require all tasks be from the same build?